BTree.cc

```diff
@@ -18,26 +19,599 @@ using std::cout;
 #include "BTree.h"
 #include "BTreeFile.h"
 #include "BTreeBlock.h"
+#include <cmath>
+#include <stack>
 
 BTree::BTree(string name)
 : _file(* new BTreeFile(name))
 { }
 
 #ifndef PROFESSOR_VERSION
 
+
+/*
+ * Recursive auxiliary function for finding a block containing a key
+ * @param1 string specifying key to be found
+ * @param2 BTreeBlock number of root to start serach from
```

Comment:
Caller's comment says param2 is modified in an important way.  Definitely describe that here.